packages/http/src/validator/validators/__tests__/body.spec.ts

```diff
@@ -169,6 +169,111 @@ describe('validate()', () => {
       assertRight(validate({ name: 'Item One', title: 'title' }, specs, ValidationContext.Output, 'application/json'));
     });
   });
+
+  describe('merge allOf', () => {
```

Comment:
should we add a test for mutually exclusive constraints imposed by `allOf` members?

Reply:
I made [some notes about mutually exclusive `allOf`](https://app.zenhub.com/workspaces/team-platinum-falcons-5eb9bc4a1fddfe30b94f78b1/issues/stoplightio/prism/1920#issuecomment-984065953), including a draft test that would fail.

I brought this issue up with Team P.F. this morning during stand up.  @ryotrellim seems OK with solving this problem in some other way, addressing it elsewhere in the workflow.  To me, it seems odd to include input/output validation messages that concern the _schema against which we're validating_, rather than the relationship between the input/output and the schema.

What do you think, @P0lip ?

Reply:
Admittedly I'm not fully comprehending what's happening here so lmk if we're introducing some bad UX or excessive tech debt in the context of this bug fix.

Reply:
tl;dr let's not worry about it too much now. We can do it later. We have here is already good enough.

> To me, it seems odd to include input/output validation messages that concern the schema against which we're validating

I totally agree. We should validate it earlier before we even get to this point.

@EdVinyard @ryotrellim 

>  or excessive tech debt in the context of this bug fix.

I'm not a Prism expert, so I'm not quite sure where that logic should live and cannot quite say whether it'd result in a tech debt, but I'd hope it's a rather not an overly complicated thing to address in the future.

>  introducing some bad UX

I'm mostly concerned about that part, but frankly speaking, it really boils down to how we want to handle corner cases (that unfortunately oftentimes exist in those large enterprise-grade documents).

Consider the following
```yaml
openapi: 3.0.0
paths:
  /test:
    post:
      responses:
        201:
          content:
            application/json:
              schema:
                type: object
                properties:
                  result:
                    type: string
      requestBody:
        content:
          application/json:
            schema:
              $ref: '#/components/schemas/Post'

components:
  schemas:
    Post:
      title: Post
      properties:
        id:
          type: string
        author:
          allOf:
            - readOnly: true
            - $ref: '#/components/schemas/Viewer'
            - $ref: '#/components/schemas/Guest'
      required:
        - id
        - author

    Viewer:
      title: Viewer
      type: object
      properties:
        id:
          type: integer

    Guest:
      title: Guest
      type: object
      properties:
        id:
          type: number
```

Running the very same command that Sergey posted in the ticket leads to the same error

```
 curl -X POST http://127.0.0.1:4010/test  -v -H "Content-Type: application/json" -d '{"id": "dqdd"}'

{"type":"https://stoplight.io/prism/errors#UNPROCESSABLE_ENTITY","title":"Invalid request","status":422,"detail":"Your request is not valid and no HTTP validation response was found in the spec, so Prism is generating this error for you.","validation":[{"location":["body"],"severity":"Error","code":"required","message":"must have required property 'author'"}]}* Closing connection 0
```

and it might be not apparent why it still doesn't work.

The library itself prints such an error
![image](https://user-images.githubusercontent.com/9273484/144478667-2e6b796d-c1f4-4879-8c9b-cf7c71866b63.png)
but we don't expose it to the end-user.

Either way, that's rather a nice to have kind of a thing rather than a huge defect that hampers the UX.